clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1139,7 +1139,8 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
     } else if (!ArgMD) {
       DepFile = "-";
     } else if (ArgMD->getOption().matches(options::OPT_MMD) &&
-               Args.hasArg(options::OPT_fintelfpga)) {
+               Args.hasArg(options::OPT_fintelfpga) &&
+               !Args.hasArg(options::OPT_c)) {
```

Comment:
Not sure if I understand this correctly. Could you please clarify the scenario covered hereby?

Reply:
This covers the case in which we are naming the output object: `clang++ -fsycl -fintelfpga a.cpp -c -o output.o`.  Since the bundling step is finding the dependency file based on output name, we shouldn't force the dependency file name to match the source, but rather allow for the natural setting of `-o` to take over to name the dependency file.